plugins/module_utils/foreman_helper.py

```diff
@@ -89,6 +95,75 @@ class NoEntity(object):
     pass
 
 
+class RequestResponse(object):
+    def __init__(self, resp):
+        self._resp = resp
+
+    @property
+    def status_code(self):
+        if hasattr(self._resp, 'status'):
+            status = self._resp.status
+        else:
+            status = self._resp.getcode()
+        return status
+
+    @property
+    def headers(self):
+        return self._resp.headers
+
+    def raise_for_status(self):
+        http_error_msg = ""
+        if self.status_code > 400:
+            http_error_msg = "you failed"
+        if http_error_msg:
+            raise Exception(http_error_msg)
+
+    def json(self, **kwargs):
+        return json.loads(to_native(self._resp.read()), **kwargs)
+
+
+class RequestSession(Request):
+    @property
+    def auth(self):
+        return (self.url_username, self.url_password)
+
+    @auth.setter
+    def auth(self, value):
+        self.url_username, self.url_password = value
+        self.force_basic_auth = True
+
+    @property
+    def verify(self):
+        return self.validate_certs
+
+    @verify.setter
+    def verify(self, value):
+        self.validate_certs = value
+
+    def request(self, method, url, **kwargs):
+        validate_certs = kwargs.pop('verify', None)
+        params = kwargs.pop('params', None)
+        if params:
+            url += '?' + six.moves.urllib.parse.urlencode(params)
```

Comment:
Doesn't Ansible's HTTP client code itself implement this?

Reply:
So far, nope